node/src/mempool.rs

```diff
@@ -332,13 +332,12 @@ impl MempoolSrv {
             .conf
             .mempool_download_redundancy
             .unwrap_or(DEFAULT_DOWNLOAD_REDUNDANCY);
+
+        let net = network.read().await;
+        net.wait_for_alive_nodes(max_peers, 3).await;
```

Comment:
Does this `wait_for_alive_nodes` hold a read lock over `network` for all its duration? This might seems to strengthen the case for an abstract `wait_for_alive_nodes` that would handle any synchronization internally

Reply:
This network rwlock is never accessed with write, this means that no one is blocked by this